bowling/src/ArgumentParser.cpp

```diff
@@ -0,0 +1,42 @@
+#include "ArgumentParser.hpp"
+
+#include <algorithm>
+
+ArgumentParser::ArgumentParser(int size, char** arguments) {
+    size_ = size;
+    fillArgumentsVector(arguments);
+}
+
+void ArgumentParser::fillArgumentsVector(char** arguments) {
+    arguments_.reserve(size_);
+    for (int i = 0; i < size_; ++i) {
+        arguments_.push_back(arguments[i]);
+    }
+}
+
+std::string ArgumentParser::getAppName() const {
+    return getExistingArgument(0); 
+}
+
+std::string ArgumentParser::getInputDirectory() const {
+    return getExistingArgument(1); 
+}
+
+std::string ArgumentParser::getOutputFileName() const {
+    return getExistingArgument(2); 
+}
+
+bool ArgumentParser::isHelpNeeded() const {
+     return size_ == 1 || size_ > 3 || argumentExists("-h") || argumentExists("--help");
```

Comment:
There's no difference if program is executed with -h option or without any arguments so the option -h seems useless (help is displayed anyway). Unless when executing without any arguments it shows only short description of its purpose and offers -h option for more details. If so the 'size_ == 1' condition is  redundant. But it's really nothing and can be left as it is.